v2/tools/generator/internal/codegen/pipeline/exported_properties.go

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import "github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+
+// ExportedProperties is a map of string property name to a chain of properties. For allowing the export of a property "foo" that writes
+// the status.a.b.c property to a config map would be "foo" -> []{status, a, b, c}
+type ExportedProperties = map[string][]*astmodel.PropertyDefinition
```

Comment:
Can you use (or generalize) PropertyChain here?

Reply:
Yes, this is a `propertyChain`. It's already basically generalized enough I believe, but will fix up some comments and use it.

Reply:
Actually, this is not quite true. `propertyChain` is structured in such a way as to allow us to remove certain items from the path it generates. That is needed for the context where we are generating minimal index method names (we want as few properties in the path as we can fit while still making the method unique).

In this instance, the properties will actually be used in code (`a.b.c.d`) so there's no need to remove properties from the chain.

We could still use `propertyChain` here (it doesn't hurt), but it's defined in `pipeline` and we eventually pass this map to something in `astmodel/functions`. We can't do that with `propertyChain` so we'd end up having to call `chain.properties()` on it and then rebuild the map and pass it, which generates a good bit of unneeded boilerplate. Since the actual usage of the map is in the `astmodel/functions` helper, it doesn't get us anything to use `propertyChain` unless we can pass it everywere.

A solution to this would be to promote `propertyChain` to a first class citizen in `astmodel`. I'm 50-50 as to if it's worth doing that for this case because we're not getting a lot of advantage out of using it here, and if we do feel doing that is worth it, I am not sure we should do it in this PR as it'll be a pretty big refactor of `propertyChain` usage and this PR is already pretty large.

Thoughts?

Reply:
Sounds like it should stay the way it is. We _might_ want to refactor it as you suggest, but certainly not in this PR.